b2/raw_api.py

```diff
@@ -440,6 +441,38 @@ def update_bucket(
             **kwargs
         )
 
+    def unprintable_to_question(self, string):
+        """Replace unprintable chars with a question mark."""
+        return re.sub(r'[^\x20-\x7E]', '?', string)
+
+    def check_b2_filename(self, filename):
+        """
+        Raise an appropriate exception with details if the filename is unusable.
+
+        See https://www.backblaze.com/b2/docs/files.html for the rules.
+
+        :param filename: A proposed filename in utf-8.
+        :return: None if the filename is usable."""
+        if len(filename) < 1:
+            raise UnusableFileName("Filename must be at least 1 character.")
+        if len(filename) > 1024:
+            raise UnusableFileName("Filename must be at most 1025 characters.")
+        lowest_unicode_value = ord(min(filename.decode('utf-8', 'ignore')))
+        if lowest_unicode_value < 32:
+            message = "Filename \"{}\" contains code {}, which is less than 32.".format(
+                self.unprintable_to_question(filename), lowest_unicode_value)
```

Comment:
I'd be in favor of showing the hex value, maybe using `\xAB` notation.

Reply:
That was kind of what I was leaning towards, but wasn't sure if that was too fussy.  `ls` on my Mac just prints a `?` which isn't as helpful.  Anyway, I'll go ahead with the hex.